src/lib/gui/widgets/FingerprintPreview.cpp

```diff
@@ -12,57 +12,36 @@
 
 #include <net/SecureUtils.h>
 
-FingerprintPreview::FingerprintPreview(QWidget *parent, const QList<Fingerprint> &fingerprints) : QFrame(parent)
+FingerprintPreview::FingerprintPreview(QWidget *parent, const Fingerprint &fingerprint) : QFrame(parent)
 {
   setFrameShape(QFrame::StyledPanel);
   setFrameStyle(QFrame::Sunken);
   setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Maximum);
 
-  QString sha1String;
-  QString sha256String;
-  QString sha256Art;
-
-  for (const auto &fingerprint : fingerprints) {
-    switch (fingerprint.type) {
-    case Fingerprint::Type::SHA1:
-      sha1String = deskflow::formatSSLFingerprint(fingerprint.data);
-      break;
-
-    case Fingerprint::Type::SHA256:
-      sha256String = deskflow::formatSSLFingerprintColumns(fingerprint.data);
-      sha256Art = deskflow::generateFingerprintArt(fingerprint.data);
-      break;
-    default:
-      qWarning() << "unknown fingerprint type:" << fingerprint.type;
-      break;
-    }
-  }
-
-  auto labelSha1 = new QLabel(QStringLiteral("SHA1:"), this);
-  labelSha1->setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Maximum);
-
-  auto lblSha1 = new QLabel(sha1String, this);
-  lblSha1->setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Maximum);
-  lblSha1->setTextInteractionFlags(Qt::TextSelectableByMouse);
-
-  auto sha1Layout = new QHBoxLayout();
-  sha1Layout->addWidget(labelSha1);
-  sha1Layout->addWidget(lblSha1);
-
-  auto frameSha1 = new QFrame(this);
-  frameSha1->setFrameShape(QFrame::StyledPanel);
-  frameSha1->setFrameStyle(QFrame::Sunken);
-  frameSha1->setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Maximum);
-  frameSha1->setLayout(sha1Layout);
+  setLayout(fingerprint.type == Fingerprint::Type::SHA256 ? sha256Layout(fingerprint) : emptyLayout());
 
+  adjustSize();
+  setFixedSize(size());
+}
+
+QLayout *FingerprintPreview::emptyLayout()
+{
+  auto *label = new QLabel(tr("Invalid Hash format"));
```

Comment:
Why the random caps?
```suggestion
  auto *label = new QLabel(tr("Invalid hash format"));
```